data_structures/src/chain.rs

```diff
@@ -2559,15 +2586,48 @@ impl AltKeys {
     pub fn insert_keys_from_block(&mut self, block: &Block) {
         // Add miner bn256 public keys
         if let Some(value) = block.block_header.bn256_public_key.clone() {
-            self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+            // If the provided key is valid, insert it in altkeys
+            if value.is_valid() {
+                self.insert_bn256(block.block_header.proof.proof.pkh(), value);
+            } else {
+                log::warn!(
+                    "Ignoring invalid bn256 public key {:02x?} inserted by PKH {:02x?} in the block",
+                    value, block.block_header.proof.proof.pkh()
+                 );
+            }
         }
         // Add bn256 public keys from commitment transactions
         for commit in &block.txns.commit_txns {
             if let Some(value) = commit.body.bn256_public_key.clone() {
-                self.insert_bn256(commit.body.proof.proof.pkh(), value);
+                // If the provided key is valid, insert it in altkeys
+                if value.is_valid() {
+                    self.insert_bn256(commit.body.proof.proof.pkh(), value);
+                } else {
+                    log::warn!(
+                        "Ignoring invalid bn256 public key {:02x?} inserted by PKH {:02x?} in the commit",
+                        value, block.block_header.proof.proof.pkh()
+                    );
+                }
             }
         }
     }
+
+    pub fn get_rep_ordered_bn256_list(
+        &self,
+        reputation_set: &TotalReputationSet<PublicKeyHash, Reputation, Alpha>,
+    ) -> Vec<Bn256PublicKey> {
+        let mut pkhs: Vec<PublicKeyHash> = self.bn256.iter().map(|(pkh, _)| *pkh).collect();
+        pkhs.sort();
+
+        let ordered_alts: Vec<_> = pkhs
+            .into_iter()
+            .enumerate()
+            .sorted_by_key(|(_, pkh)| reputation_set.get(pkh))
```

Comment:
Here the reputation is sorted from lower to higher, but I thought it is used to stop when we reach 2/3 of reputation. So...  would be it better sort from high to low??

Reply:
it does not really matter I guess, as long as you also compute low to high in Ethereum